index.js

```diff
@@ -42,3 +48,7 @@ module.exports = function (func, cb) {
     return deferred.promise;
   }
 };
+
+module.exports.cb = function (func, cb) {
+  return runAsync(func, cb, true);
```

Comment:
I think this could be done much simpler, and without polluting the existing code:

```js
module.exports.cb = function (func, cb) {
  return module.exports(function () {
    var args = Array.prototype.slice(arguments);
    if (args.length === func.length - 1) {
      args.push(this.async());
    }
    func.apply(this, args);
  }, cb);
};